server/aws-lsp-yaml-json/src/language-server/yamlJsonServer.ts

```diff
@@ -59,9 +80,56 @@ export const YamlJsonServerFactory =
             })
         }
 
+        const onDidOpenTextDocumentHandler = async (params: DidOpenTextDocumentParams): Promise<any> => {
+            let textDocument = await workspace.getTextDocument(params.textDocument.uri)
+            if (!textDocument) {
+                textDocument = TextDocument.create(
+                    params.textDocument.uri,
+                    params.textDocument.languageId,
+                    params.textDocument.version,
+                    params.textDocument.text
+                )
+            }
+
+            const diagnostics = await service.doValidation(textDocument)
+
+            await lsp.publishDiagnostics({
+                uri: params.textDocument.uri,
+                diagnostics: diagnostics,
+            })
+        }
+
+        const onFormatHandler = async (params: DocumentFormattingParams): Promise<TextEdit[] | null> => {
+            const getTextDocumentFullRange = (textDocument: TextDocument): Range => {
+                return {
+                    start: {
+                        line: 0,
+                        character: 0,
+                    },
+                    end: {
+                        line: textDocument.lineCount,
+                        character: 0,
```

Comment:
does this mean we exclude the last line because we set end index to character 0?

Reply:
last line has index "textDocument.lineCount - 1". line: textDocument.lineCount is intentional here. 
This range include full document content